### PR TITLE
Artwork contact_message should not expose availability when availability is hidden

### PR DESCRIPTION
**Pull request by oxaudo**
Should resolve https://artsyproduct.atlassian.net/browse/EV-133

### CODE REVIEW

**Comment by oxaudo** [comment]
@alloy -Btw... not sure if this issue will resolve itself once we remove prefilled message from artwork page. I believe the same `contact_message` is used in inquiry pop up.

File: src/schema/artwork/index.js [hunk 1]
```diff
@@ -152,13 +152,16 @@ export const artworkFields = () => {
...
       description: "Pre-filled inquiry text",
-      resolve: ({ partner, availability }) => {
+      resolve: ({ partner, availability_hidden, availability }) => {
         if (partner && partner.type === "Auction") {
           return [
             "Hello, I am interested in placing a bid on this work.",
             "Please send me more information.",
           ].join(" ")
         }
+        if (availability_hidden) {
```
Comment (mzikherman) [inline src/schema/artwork/index.js]: This seems sort of non-sensical, but maybe this conditional should go first? In this case, a work that is at an auction (but marked to be 'hidden'), will still include that auction-specific text.

So, just to be safe/thorough, maybe we move this up before that?

**Comment by mzikherman** [comment]
Good catch! I suppose we have works that _are_ inquireable (so should be showing the 'Contact' links/button), but that have hidden availabilities.

I believe for artwork brick contexts, we use `sale_message`, which _does_ take that into account. So, nice catch that we were leaking the information in this context! 👍

File: src/schema/artwork/index.js (hunk 1 above)
Comment (oxaudo) [reply src/schema/artwork/index.js]: I actually thought about that and kinda decided that auction takes precedence. Like if the artwork is from auction house - that means it should definitely be biddable one way or the other.... Though maybe that is not the case. I'm happy to move the check up if you still think it's more logical-:)

Or I guess we could ask somebody from #product what should happen when artwork is from Auction partner and still has hidden availability (just typing it - it should be very unlikely case:))

Comment (alloy) [reply src/schema/artwork/index.js]: > Or I guess we could ask somebody from #product what should happen when artwork is from Auction partner and still has hidden availability (just typing it - it should be very unlikely case:))

What was the outcome of this?

Comment (oxaudo) [reply src/schema/artwork/index.js]: Ok, @mzikherman. @alloy - the answer to above question is that the flow that allows inquiries on works from Auction partners is completely deprecated in current UI so it's truly does not matter what takes precedence here.

Ideally we should remove the code that takes this flow into account across the board, but that should probably be a separate ticket either way.

This PR should be good to merge as is.

**Comment by alloy** [comment]
Alright, I’ll take this 👍

**Pull request merged**